core/src/main/scala/org/apache/spark/HeartbeatReceiver.scala

```diff
@@ -128,14 +128,16 @@ private[spark] class HeartbeatReceiver(sc: SparkContext, clock: Clock)
 
     // Messages received from executors
     case heartbeat @ Heartbeat(executorId, accumUpdates, blockManagerId, executorUpdates) =>
+      var reregisterBlockManager = !sc.isStopped
```

Comment:
This sounds like a race condition that happens during SparkContext shutdown. So it's also possible that the SparkContext is stopped right after we sent the `HeartbeatResponse`. In that case, IIUC, the issue will still exist.

Does the current behavior cause any real issue?

Reply:
Thank. you for review, @Ngone51 . That's true. In the production environment, I hit those intermediate status. And, this will help us simplify the situation.

For the following, yes. We are After the apps `sc.stop` takes a longer time than we expect.
> Does the current behavior cause any real issue?